core/src/main/java/org/apache/iceberg/util/ArrayUtil.java

```diff
@@ -59,4 +68,221 @@ private ArrayUtil() {
       return null;
     }
   }
+
+  /**
+   * <p>Converts an array of object Booleans to primitives.
+   *
+   * <p>This method returns {@code null} for a {@code null} input array.
+   *
+   * @param array  a {@code Boolean} array, may be {@code null}
+   * @return a {@code boolean} array, {@code null} if null array input
+   * @throws NullPointerException if array content is {@code null}
+   */
+  public static boolean[] toPrimitive(final Boolean[] array) {
+    if (array == null) {
+      return null;
+    } else if (array.length == 0) {
+      return EMPTY_BOOLEAN_ARRAY;
+    }
+    final boolean[] result = new boolean[array.length];
+    for (int i = 0; i < array.length; i++) {
+      result[i] = array[i].booleanValue();
+    }
+    return result;
+  }
```

Comment:
If the code for these comes from `org.apache.commons:commons-lang3`, it should be credited as such some where in the file.

Reply:
Thanks for review. Updated the comments.